nbs/directives.ipynb

```diff
@@ -0,0 +1,828 @@
+   "source": [
+    "Example:\n",
+    "\n",
+    "```python\n",
+    "#|eval: false\n",
+    "raise Exception(\"I'm not raised because I'm not run\")\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#|eval: false\n",
+    "raise Exception(\"I'm not raised because I'm not run\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Compared to:\n",
+    "\n",
+    "```python\n",
+    "#|eval: true\n",
+    "assert 1 + 1 == 2\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#|eval: true\n",
+    "assert 1 + 1 == 2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## `#|echo: <true|false>`\n",
+    "\n",
+    "This makes sure that only the output of a code cell is shown, not its input.\n",
+    "\n",
+    "```python\n",
+    "#|echo: false\n",
+    "print('you can see the output but not the code!')\n",
+    "```\n",
+    "\n",
+    "which results in:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "you can see the output but not the code!\n"
+     ]
+    }
+   ],
+   "source": [
+    "#|echo: false\n",
+    "print('you can see the output but not the code!')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Compared to:\n",
+    "```python\n",
+    "#|echo: true\n",
+    "print('you can see the code and the output!')\n",
+    "```\n",
+    "which results in:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "you can see the code and the output!\n"
+     ]
+    }
+   ],
+   "source": [
+    "#|echo: true\n",
+    "print('you can see the code and the output!')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## `#|hide_line`\n",
+    "\n",
+    "You can use this to hide as specific line in your code.  For example:\n",
+    "\n",
+    "```python\n",
+    "def _secret(): ...\n",
+    "\n",
+    "for i in range(3):\n",
+    "    _secret() #|hide_line\n",
+    "    print(i)\n",
+    "```\n",
+    "\n",
+    "becomes this:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "0\n",
+      "1\n",
+      "2\n"
+     ]
+    }
+   ],
+   "source": [
+    "def _secret(): ...\n",
+    "\n",
+    "for i in range(3):\n",
+    "    _secret() #|hide_line\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## `#|output: <true|false|asis>`\n",
```

Comment:
This is handled by quarto, not nbdev, so I'd suggest removing these docs. Instead, you might want to have a section that mentions what quarto directives are, lists a couple of useful ones, and links to the quarto docs about them.

Reply:
@dleen I talked to Jeremy about this.  I personally find this one-page cheat sheet of all the most important nbdev + quarto directives incredibly valuable.  This is what we settled on:

1. Have a section up front called "nbdev vs Quarto directives" that explains a bit about what directives are, and how Quarto has many of them, and nbdev adds additional ones.  Also mention that we will denote nbdev specific directives with an emoji like 📔 and quarto ones with an emoji like ❤️   .  Feel free to use different emojis if you are more creative than me!  
2. Categorize the directives as much as possible by "functionality" ex: (1) cell visibility (2) testing (3) exporting source code (4) etc ....
3. When you do write about quarto directives, do not write more than a couple of sentences about what they do, and link to the appropriate documentation in quarto in https://quarto.org/ (LMK if you need help finding any links) just because we don't want to duplicate the quarto docs too much

Reply:
Tried my best here but probably didn't delete enough. See https://dleen.github.io/nbdev/explanations/directives.html